lib/fetch/index.js

```diff
@@ -126,8 +127,49 @@ class Fetch extends EE {
   }
 }
 
+// hoisting subscribers variables
+let subscribers
+
+// publishIfSubscribed will check if there are subscribers to the channel
+// add set additional context before publishing the event
+function publishIfSubscribed (name, context, type, load) {
+  if (subscribers[name]) {
+    if (type) context[type] = load
+    channels[name].publish(context)
+    if (type) context[type] = null
+  }
+}
```

Comment:
You could actually simplify it:

```js
function publishIfSubscribed (name, context, error) {
  if (subscribers[name]) {
    if (error) context.error = error
    channels[name].publish(context)
    if (error) context.error = null
  }
}
```

Reply:
I see, since I am only using type/load when I have an error. The original use case was to use it for all occurrences, such as with result, however I could not, given when I would need to set the result. I will update 👍🏼